src/Data/Singletons/Internal.hs

```diff
@@ -97,6 +111,40 @@ class SingKind k where
 data SomeSing k where
   SomeSing :: Sing (a :: k) -> SomeSing k
 
+-- | A pattern synonym for going between a term and the corresponding
+--
+-- As a constructor: this takes a singleton to its demoted (base)
+-- type.
+--
+-- >>> :t AsSing @Bool
```

Comment:
IIRC, I believe you need to escape the `@` character (as `\@`) in order for Haddock to render it correctly.

Reply:
Hm Kmett seems to use `\@` all over the place but it seems to work withou it here https://hackage.haskell.org/package/singletons-2.3.1/docs/Data-Singletons.html#v:singFun1